# 异常

```
class MyExp:Exception
{
    public MyExp(string \_msg) : base(_msg) { }
}

void Fun()
{
    try
    {
        double y = Convert.ToDouble("xx");
        Console.WriteLine(y);
    }catch(Exception e)
    {
        throw new MyExp("err");
    }
}


try
{
    Fun();
}
catch (MyExp e)
{
    Console.WriteLine(e.Message);
}
catch (Exception e)
{
    Console.WriteLine(e.ToString());
}
```
